### PR TITLE
sci-libs/vtk: add 9.3.1, cgns, minimal and netcdf USE flags

### PR DESCRIPTION
**Pull request by cjmayo**
Closes: https://bugs.gentoo.org/939045

---

The first commit is mainly a rename, dropping an obsolete CMake variable:

QA Notice: One or more CMake variables were not used by the project:
   VTK_MODULE_USE_EXTERNAL_VTK_ioss

and patches:

  vtk-9.3.0-pdal-2.6.0.patch
  https://gitlab.kitware.com/vtk/vtk/-/commit/ccee5db79d6037c4d752b5f29dc30fb0dfd039bf
  
  vtk-9.3.0-too-many-initializer-values.patch was fixed slighty differently in:
  https://gitlab.kitware.com/vtk/vtk/-/commit/6d0ed05f2afe80f61b2c38a57da1d4a58d87336e
  
  vtk-9.3.0-fix-Java-error-with-integer-enum-overloads.patch
  https://gitlab.kitware.com/vtk/vtk/-/commit/dd7acfb4548dc76f1191aa96778f35c0cda80d35
  
  vtk-9.3.0-core-octree_node.txx.patch was fixed before
  https://gitlab.kitware.com/vtk/vtk/-/commit/65fc526a83ac829628a9462f61fa57f1801e2c7e
  
  vtk-9.3.0-r1-expat-2.6.0.patch
  https://gitlab.kitware.com/vtk/vtk/-/commit/3efa07ad277efe5c1d11a2ef2b907c095f68bbef

---

The minimal configuration is based on 
https://public.kitware.com/pipermail/vtkusers/2015-May/090922.html

with some updating for changes since 2015

These flags allow a smaller build, removing the need for the less common and X11 dependencies. It's not that minimal. It is sufficient to build FreeCAD 1.0RC2.

cgns and/or netcdf can be used with minimal.

---

First commit changes:

```diff
--- vtk-9.3.0-r3.ebuild
+++ vtk-9.3.1.ebuild
@@ -38,7 +38,7 @@
 
 LICENSE="BSD LGPL-2"
 SLOT="0/${MY_PV}"
-KEYWORDS="amd64 ~arm ~arm64 ~x86 ~amd64-linux ~x86-linux"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86 ~amd64-linux ~x86-linux"
 
 # TODO: Like to simplifiy these. Mostly the flags related to Groups.
 IUSE="all-modules boost cuda debug doc examples ffmpeg +freetype gdal gles2-only imaging
@@ -159,15 +159,10 @@
 
 PATCHES=(
 	"${FILESDIR}/${PN}-9.2.5-pegtl-3.x.patch"
-	"${FILESDIR}/${PN}-9.3.0-pdal-2.6.0.patch"
-	"${FILESDIR}/${PN}-9.3.0-too-many-initializer-values.patch"
 	"${FILESDIR}/${PN}-9.3.0-java.patch"
 	"${FILESDIR}/${PN}-9.3.0-opencascade.patch"
-	"${FILESDIR}/${PN}-9.3.0-fix-Java-error-with-integer-enum-overloads.patch"
 	"${FILESDIR}/${PN}-9.3.0-ThrustPatches.patch"
-	"${FILESDIR}/${PN}-9.3.0-core-octree_node.txx.patch"
 	"${FILESDIR}/${PN}-9.3.0-ThirdParty-gcc15.patch"
-	"${FILESDIR}/${PN}-9.3.0-r1-expat-2.6.0.patch"
 )
 
 DOCS=( CONTRIBUTING.md README.md )
@@ -260,7 +255,7 @@
 #	and can currently not unbundled:
 #	diy2, exodusII, fides, h5part, kissfft, loguru, verdict, vpic,
 #	vtkm, xdmf{2,3}, zfp
-# TODO: exprtk, ioss
+# TODO: exprtk
 # Note: As of v9.2.2 we no longer drop bundled libraries, when using system
 # libraries. This just saves a little space. CMake logic of VTK on ThirdParty
 # libraries avoids automagic builds, so deletion is not needed to catch these.
@@ -368,7 +363,6 @@
 		# not packaged in Gentoo
 		-DVTK_MODULE_USE_EXTERNAL_VTK_fast_float=OFF
 		-DVTK_MODULE_USE_EXTERNAL_VTK_exprtk=OFF
-		-DVTK_MODULE_USE_EXTERNAL_VTK_ioss=OFF
 		-DVTK_MODULE_USE_EXTERNAL_VTK_verdict=OFF
 
 		-DVTK_RELOCATABLE_INSTALL=ON
```

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [ ] I have run `pkgcheck scan --commits --net` to check for issues with my commits.
- Added with `pkgdev commit`

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: sci-libs/vtk/vtk-9.3.1.ebuild [hunk 1]
```diff
@@ -388,6 +392,35 @@ src_configure() {
 		-DVTK_WRAP_PYTHON="$(usex python)"
 	)
 
+	if use minimal; then
```
Comment (Nowa-Ammerlaan) [inline sci-libs/vtk/vtk-9.3.1.ebuild:395]: This is a bit confusing to me, `use minimal` is enabling a lot of extra options. Naively I would expect this the other way around, i.e. `use minimal` disabling a bunch of options.

File: sci-libs/vtk/vtk-9.3.1.ebuild [hunk 2]
```diff
@@ -41,19 +41,21 @@ SLOT="0/${MY_PV}"
 KEYWORDS="~amd64 ~arm ~arm64 ~x86 ~amd64-linux ~x86-linux"
 
 # TODO: Like to simplifiy these. Mostly the flags related to Groups.
-IUSE="all-modules boost cuda debug doc examples ffmpeg +freetype gdal gles2-only imaging
-	java las +logging mpi mysql odbc opencascade openmp openvdb pdal postgres python qt5
-	qt6 +rendering sdl tbb test +threads tk video_cards_nvidia +views vtkm web"
+IUSE="all-modules boost +cgns cuda debug doc examples ffmpeg +freetype gdal gles2-only imaging
+	java las +logging minimal mpi mysql +netcdf odbc opencascade openmp openvdb pdal postgres
+	python qt5 qt6 +rendering sdl tbb test +threads tk video_cards_nvidia +views vtkm web"
```
Comment (Nowa-Ammerlaan) [inline sci-libs/vtk/vtk-9.3.1.ebuild:46]: We should consider dropping `qt5` in this release, the latest freecad rc works reasonably well with qt6 and as I remember that was the only reverse dependency.

In any case, we should definitely drop this auto-magic behaviour:
```
use qt6 && use qt5 && ewarn "Both qt5 and qt6 USE flags have been selected. Using qt5!"
```
so we can get rid of the weird `sci-libs/vtk[qt6,-qt5]` dependency in the freecad ebuild.

**Comment by negril** [comment]
@Nowa-Ammerlaan I have a whole other set of changes on top of this that we need before we can merge this.

**Comment by negril** [comment]
Oh I will happily build upon your work. That was just letting the proxy maintainers know that there is more to come.

You don't happen to be on IRC?

**Pull request closed without merging**